FfaFeaturePkg/Library/NotificationServiceLib/NotificationServiceLib.c

```diff
@@ -342,25 +313,30 @@ DestroyHandler (
   Service   = NULL;
   ReturnVal = NOTIFICATION_STATUS_INVALID_PARAMETER;
 
-  /* Extract the UUID from the message */
-  NotificationServiceExtractUuid (Request, Uuid);
+  /* Extract the UUID from the message x7-x8 (i.e. Arg3-Arg4) */
+  NotificationServiceExtractUuid (Request->Arg3, Request->Arg4, Uuid);
 
   /* Attempt to find the UUID within our list */
   for (Index = 0; Index < NOTIFICATION_MAX_SERVICES; Index++) {
-    if (!CompareMem (Uuid, NotificationServices[Index].ServiceUuid, sizeof (Uuid))) {

```

Comment:
Searching the NotificationServices to get a pointer to the service is duplicated a few times in this file. Consider creating a helper. e.g. `NotifService* FindNotificationService(UINT8 Uuid[16]);`